app/src/main/java/org/dicio/dicio_android/input/InputDevice.java

```diff
@@ -33,6 +35,12 @@
          */
         void onInputReceived(String input);
 
+        /**
+         * Called when alternative Inputs are Received from Vosk
+         * Alternative Inputs are tried, when the first doesn't match
```

Comment:
This is an abstract class, so the documentation should not talk about Vosk, which is just one possible implementation
```suggestion
         * Called when some input was received from the user, and there are many possible alternative inputs 
         * @param input the list of alternative received inputs, sorted by confidence (the most confident item is the first one)
```